Directory.Build.targets

```diff
@@ -0,0 +1,194 @@
+    <UnityManagedPath>$(UnityRoot)\Data\Managed</UnityManagedPath>
+    <UnityExec>$(UnityRoot)\Unity.exe</UnityExec>
+    <StandalonePlayerName>buildWindows64Player</StandalonePlayerName>
+    <ArtifactName>game.exe</ArtifactName>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$([MSBuild]::IsOSPlatform('OSX'))">
+    <UnityRoot Condition="Exists('/Applications/Unity/Hub/Editor/$(UnityVersion)/Unity.app/Contents/Managed/UnityEngine.dll')">/Applications/Unity/Hub/Editor/$(UnityVersion)/Unity.app/</UnityRoot>
+    <!--If not using Unity Hub, tries to pick whatever Unity version is installed on the machine-->
+    <UnityRoot Condition="$(UnityRoot) == '' AND Exists('/Applications/Unity/Unity.app/Contents/Managed/UnityEngine.dll')">/Applications/Unity/Unity.app/</UnityRoot>
+    <UnityManagedPath>$(UnityRoot)/Contents/Managed</UnityManagedPath>
+    <UnityExec>$(UnityRoot)/Contents/MacOS/Unity</UnityExec>
+    <StandalonePlayerName>buildMacOS64Player</StandalonePlayerName>
+    <ArtifactName>game</ArtifactName>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <UnitySampleProjectPath>$(MSBuildProjectDirectory)/../../samples/unity-of-bugs</UnitySampleProjectPath>
+    <UnityTestPlayModeResultFilePath>../../artifacts/test/playmode/results.xml</UnityTestPlayModeResultFilePath>
+    <UnityTestEditModeResultFilePath>../../artifacts/test/editmode/results.xml</UnityTestEditModeResultFilePath>
+  </PropertyGroup>
+
+  <Target Name="CleanUnityTestResults" AfterTargets="Clean">
+    <Delete Files="$(UnityTestPlayModeResultFilePath)" />
+    <Delete Files="$(UnityTestEditModeResultFilePath)" />
+  </Target>
+
+  <!-- If Unity Libraries don't exist, run a Unity build -->
+  <Target Name="UnityRestorePackages"
+    Condition="!Exists('$(UnitySampleProjectPath)/Library/ScriptAssemblies/UnityEngine.TestRunner.dll')"
+    BeforeTargets="BeforeBuild">
+    <Error Condition="$(UnityRoot) == ''" Text="Couldn't find Unity."></Error>
+
+    <!-- Even with a successful build, Unity will error on 'usbmuxd' or log out to std-error which breaks msbuild.
+    We need to run a unity build to restore the test packages and for that reason we'll ignore errors here and assume a later step will validate the build is actually working:
+      The offending error:
+      [usbmuxd] Stop listen thread
+      [usbmuxd] Error:
+      [usbmuxd] Listen thread exitingit -batchmode -nographics -logFile - -projectPath $(UnitySampleProjectPath) -$(StandalonePlayerName) $(ArtifactName)
+    Related: https://forum.unity.com/threads/6572-debugger-agent-unable-to-listen-on-27.500387/  -->
+    <Exec Command="&quot;$(UnityExec)&quot; -quit -batchmode -nographics -logFile - -projectPath $(UnitySampleProjectPath) -$(StandalonePlayerName) $(ArtifactName) "
+      IgnoreStandardErrorWarningFormat="true"
+      IgnoreExitCode="true"></Exec>
+  </Target>
+
+  <!-- Run PlayMode tests with dotnet test -->
```

Comment:
Does it actually run with `dotnet test`? Looks like it runs with `dotnet msbuild /t:UnityPlayModeTest`. It's possible to make it run with `dotnet test` though if you make a custom test adapter, which is not difficult but most likely not worth it though.

Reply:
fixed